docs/integration-testing.md

```diff
@@ -19,53 +18,66 @@ typically run faster. Sometimes that's not (easily) possible, especially when an
-depend on the concrete system you're using. As a general rule of thumb long-running one-time setup should be done in
-the `@BeforeAll` so as not to extend the run-time of the test unnecessarily. In contrast, in most cases it is **not**
-advisable to deploy/provision the external system itself in either of those methods. In other words, provisioning a
-CosmosDB or spinning up a Postgres docker container directly from test code should generally be avoided, because it will
-introduce code that has nothing to do with the test and may cause security problems (privilege escalation through the
-Docker API), etc.
+- `@AzureStorageIntegrationTest`: Marks an integration test with `azure-storage-integration-test` Junit tag. This should be used when the integration test is related to `Azure Storage`.
+- `@AzureCosmosDbIntegrationTest`: Marks an integration test with `azure-cosmosdb-integration-test` Junit tag. This should be used when the integration test is related to `Azure CosmosDB`.
+- `@AwsS3IntegrationTest`: Marks an integration test with `aws-s3-integration-test` Junit tag. This should be used when the integration test is related to `AWS S3`.
+- `@DapsTest`: Marks an integration test with `daps-integration-test` Junit tag. This should be used when the integration test is related to `Daps IAM`.
 
-This does not at all discourage the use of external test environments like containers, rather, the external system
-should be deployed in the CI script (e.g. through Github's `services` feature), or there might even be a dedicated test
-instance running continuously, e.g. a CosmosDB test instance in Azure. In the latter case we need to be careful to avoid
-conflicts (e.g. database names) when multiple test runners access that system simultaneously and to properly clean-up
-any residue before and after the test.
+We encourage you to use these available annotation but if your integration test does not fit in one of these available annotations and you want to categorize them base on their technologies then feel free to create a new annotations but make sure to use composite annotations which contains `@IntegrationTest`. If you do not wish to categorize base on their technologies then you can use already available `@IntegrationTest` annotation.
+
+- By default JUnit test runner ignores all integration tests because in root `build.gradle.kts` file we have excluded all tests with `integration-test` Junit tag.
+
+All integration tests should specify annotation to categorize them and the `"...IntegrationTest"` postfix to distinguish them clearly from unit tests. They should reside in the same package as unit tests because all tests should maintain package consistency to their test subject.
+
+Any credentials, secrets, passwords, etc. that are required by the integration tests should be passed in using environment variables. A good way to access them is `ConfigurationFunctions.propOrEnv()` because then the credentials can also be supplied via system properties.
```

Comment:
spurious changes due to reformatting

Reply:
Updated it. To remove extra reformatting.